framework/global/settings.cpp

```diff
@@ -0,0 +1,225 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "settings.h"
+#include <QSettings>
+
+#include "log.h"
+
+using namespace mu::framework;
+using namespace mu::async;
+
+Settings::Settings()
+{
+#ifndef Q_OS_MAC
+    QSettings::setDefaultFormat(QSettings::IniFormat);
+#endif
+}
+
+void Settings::addItem(const Item& item)
+{
+    m_items.insert({ item.key, item });
+}
+
+void Settings::addItem(const Key& key, const Val& val)
+{
+    m_items.insert({ key, Item(key, val) });
+}
+
+const Settings::Item& Settings::findItem(const Key& key) const
```

Comment:
I am trying to understand the case when I need to search for the entire Item, and not just get its value by key. How much is it in demand?

Reply:
I think this is needed to create a settings form, with grouping, user names, etc., and not just a list.  
Plus, for the consistency of the interface, if there is a method to add an item, let there be a method and get the item :)